contrib/scripts/kind.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+# Copyright 2017-2020 Authors of Cilium
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+set -o pipefail
+
+# Sample usages
+# ./contrib/scripts/kind.sh
+# IPv6=1 ./contrib/scripts/kind.sh
+# NO_BUILD=1 ./contrib/scripts/kind.sh
+# NO_PROVISION=1 ./contrib/scripts/kind.sh
+
+# Enable IPv6 mode. By default using IPv4
+export 'IPv6'="${IPv6:0}"
+# Set NO_BUILD=1 to avoid building docker images again
+export 'NO_BUILD'="${NO_BUILD:0}"
+# Set NO_PROVISION=1 to use existing kind cluster
+export 'NO_PROVISION'="${NO_PROVISION:0}"
+
+function help() {
+  printf "Run with default values:\n\t./contrib/scripts/kind.sh\n\n"
+  printf "Using existing docker images:\n\t./contrib/scripts/kind.sh\n\n"
+  printf "Using existing cluster:\n\tNO_PROVISION=1 ./contrib/scripts/kind.sh\n\n"
+  exit 1
+}
+
+# Build cilium and cilium operator docker images
+if [[ "${NO_BUILD}" -ne "1" ]]; then
+  make -j docker-cilium-image docker-operator-generic-image
+fi
+
+# Provision kind cluster if required
+if [[ "${NO_PROVISION}" -ne "1" ]]; then
+  kind delete cluster
+  # clean up previous kind network
+  docker system prune -f
```

Comment:
If this script is intended to be used locally by users wanting to spin up a local `kind` cluster, then this line will remove a lot more things than just the previous `kind` network, e.g. build cache.